lib/taskana-core/src/main/java/pro/taskana/common/internal/jobs/JobRunner.java

```diff
@@ -25,138 +19,60 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(JobRunner.class);
   private final TaskanaEngine taskanaEngine;
   private final JobServiceImpl jobService;
-  private TaskanaTransactionProvider<Object> txProvider;
+  private TaskanaTransactionProvider txProvider;
 
   public JobRunner(TaskanaEngine taskanaEngine) {
     this.taskanaEngine = taskanaEngine;
     jobService = (JobServiceImpl) taskanaEngine.getJobService();
   }
 
-  public void registerTransactionProvider(TaskanaTransactionProvider<Object> txProvider) {
+  public void registerTransactionProvider(TaskanaTransactionProvider txProvider) {
     this.txProvider = txProvider;
   }
 
   public void runJobs() {
-    try {
-      List<ScheduledJob> jobsToRun = findAndLockJobsToRun();
-      for (ScheduledJob scheduledJob : jobsToRun) {
-        runJobTransactionally(scheduledJob);
-      }
-    } catch (Exception e) {
-      LOGGER.error("Error occurred while running jobs: ", e);
-    }
+    findAndLockJobsToRun().forEach(this::runJobTransactionally);
   }
 
   private List<ScheduledJob> findAndLockJobsToRun() {
-    List<ScheduledJob> availableJobs = jobService.findJobsToRun();
-    List<ScheduledJob> lockedJobs = new ArrayList<>();
-    for (ScheduledJob job : availableJobs) {
-      lockedJobs.add(lockJobTransactionally(job));
-    }
-    return lockedJobs;
+    return TaskanaTransactionProvider.executeInTransactionIfPossible(
+        txProvider,
+        () -> jobService.findJobsToRun().stream().map(this::lockJob).collect(Collectors.toList()));
```

Comment:
This is the most important change: fetch and lock the jobs in one transaction, so that other threads / instances cannot access the fetched jobs. 
Together with the changes in the JobMapper this resolves the issue. Everything else is refactoring/restructuring.